alerts/report_sources/goma/alerts_goma_2019-10-26.Rmd

```diff
@@ -0,0 +1,2469 @@
+legitimate Aire de Santes belonging to Goma present in the database). Note - All 
+other Aire de Santes are aggregated into the category "other".
+
+Note - All other Aire de Santes are aggregated into the category "other".
+
+**Note:** To emphasise, the ranking of top Aire de Santes is done only on those Aire 
+de Santes that legitimately belong to Goma. Those that do not belong to Goma or are
+not Aire de Santes at all, are not included when it comes to plotting the Aire de Santes
+with the most alerts. However, these alerts ARE included when it comes to plotting the
+number of alerts by Sous-Coordination or Zone de Sante (assuming they have a legitimate
+Zone de Sante inputted).
+
+For the purposes of plotting by Aire de Sante, we first have to remove (filter) all 
+instances in the data where individuals have been erroneously assigned to either
+an Aire de Sante that does not belong to Goma, or that is not an Aire de Sante. Once
+this filtering step is done, we then create a new variable using `top_values`:
+
+
+``` {r variable_creation}
+
+## bleed
+bleed <- x %>%
+  select(contains("saignement")) %>%
+  apply(1, function(e) ifelse(any(e == "oui"), "oui", "non"))
+x <- mutate(x, bleed = bleed)
+table(x$bleed, useNA = "ifany")
+
+## epiweek report
+x <- x %>%
+  mutate(epiweek_report = 
+           aweek::date2week(date, week_start = "Monday", floor_day = TRUE)) %>%
+  mutate(epiweek_report_label = 
+           aweek::week2date(date, week_start = "Monday", floor_day = TRUE)) 
+
+
+
+x <- x %>% 
+  mutate(top_zones = top_values(zone_de_sante, 5)) %>%
+  mutate(top_aires = top_values(aire_de_sante, 14))
+```
+
+One last round of dictionary-based cleaning:
+
+```{r last_cleaning}
+
+x <- x %>%
+  clean_variable_spelling(wordlists = cleaning_rules)
+
+```
+
+Remove data from before 2019-01-01 as not as reliable. 
+
+```{r remove_pre_2019}
+
+x <- x %>%
+  filter(date >= "2019-01-01")
+
+```
+
+<!-- ======================================================= -->
+## Outcomes
+
+Outcome is defined for alerts which have either been validated or invalidated.
+
+```{r outcomes}
+
+## get only known outcomes
+outcomes <- x %>%
+  filter(status %in% c("validee", "invalidee")) %>%
+  droplevels() %>% 
+  mutate(validee = 1 * (status == "validee"))
+
+```
+
+<!-- ======================================================= -->
+## Check admin areas are correct
+
+```{r load_area_names_data}
+
+## load the data
+current_spatial
+area_names <- rio::import(current_spatial) %>%
+  clean_data(guess_dates = FALSE)
+
+as_names <- unique(area_names$as)
+zs_names <- unique(area_names$zs)
+
+```
+
+### Identify unknown areas  
+
+```{r identify_area_names_data}
+
+## load the data
+n_rows <- nrow(x)
+
+unknown_as <- x %>% 
+  filter(!aire_de_sante %in% as_names)
+
+most_recent_unknown_as <- max(unknown_as$date, na.rm = T)
+
+start_date <- database_date - 20
+unknown_as_3weeks <- x %>% 
+  filter(!aire_de_sante %in% as_names & date >= start_date)
```

Comment:
remove the `&` and add a second argument to `filter` - more readable, and no operator priorities issues